calico_containers/calicoctl.py

```diff
@@ -677,6 +690,77 @@ def profile_rule_update(name):
     print "Successfully updated rules on profile %s" % name
 
 
+def profile_rule_add_remove(
+        operation,
+        name, position, action, direction,
+        protocol=None,
+        icmp_type=None, icmp_code=None,
+        src_net=None, src_tag=None, src_ports=None,
+        dst_net=None, dst_tag=None, dst_ports=None):
+    """
+    Add or remove a rule from a profile.
+
+    Arguments not documented below are passed through to the rule.
+
+    :param operation: "add" or "remove".
+    :param name: Name of the profile.
+    :param position: Position to insert/remove rule or None for the default.
+    :param action: Rule action: "allow" or "deny".
+    :param direction: "inbound" or "outbound".
+
+    :return:
+    """
+    # Convert the input into a Rule.
+    rule_dict = {k: v for (k, v) in locals().iteritems()
+                 if k in Rule.ALLOWED_KEYS and v is not None}
+    rule_dict["action"] = action
+    if (protocol not in ("tcp", "udp")) and (src_ports is not None or
+                                             dst_ports is not None):
+        print "Ports are not valid with protocol %r" % protocol
+        sys.exit(1)
+    rule = Rule(**rule_dict)
+
+    # Get the profile.
+    try:
+        profile = client.get_profile(name)
+    except KeyError:
+        print "Profile %s not found." % name
+        sys.exit(1)
+
+    if direction == "inbound":
+        rules = profile.rules.inbound_rules
+    else:
+        rules = profile.rules.outbound_rules
+
+    if operation == "add":
+        if position is None:
+            # Default to append.
+            position = len(rules)
+        if not 0 <= position <= len(rules):
```

Comment:
The --at suggests what position you are inserting it at (same as the regular python indexing).

So since our indexing starts at 1, a value of 1 would be insert at 1 and shift everything else along.